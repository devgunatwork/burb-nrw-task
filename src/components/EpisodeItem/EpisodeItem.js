import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import EPISODE_SHAPE from '../../shapes/episode.shape';
import StyledLink from '../StyledLink';

const EpisodeItem = ({ episode }) => {
  const { id, name, number, season } = episode;

  return (
    <ListItem alignItems="flex-start">
      <StyledLink to={`/episode/${id}`}>
        <ListItemText primary={`${season}.${number} ${name}`} />
      </StyledLink>
    </ListItem>
  );
};

EpisodeItem.propTypes = {
  episode: EPISODE_SHAPE.isRequired,
};

export default EpisodeItem;
