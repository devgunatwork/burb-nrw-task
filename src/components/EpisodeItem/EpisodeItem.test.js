import React from 'react';
import EpisodeItem from './EpisodeItem';

describe('components', () => {
  describe('EpisodeItem', () => {
    describe('UI', () => {
      it('should match the snapshot', () => {
        const wrapper = shallow(
          <EpisodeItem
            episode={{
              id: 123,
              url: 'test url',
              name: 'name',
              season: 1,
              number: 1,
            }}
          />
        );

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
