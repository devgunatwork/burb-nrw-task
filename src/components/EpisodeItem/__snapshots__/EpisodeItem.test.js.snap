// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components EpisodeItem UI should match the snapshot 1`] = `
<WithStyles(ForwardRef(ListItem))
  alignItems="flex-start"
>
  <StyledLink
    to="/episode/123"
  >
    <WithStyles(ForwardRef(ListItemText))
      primary="1.1 name"
    />
  </StyledLink>
</WithStyles(ForwardRef(ListItem))>
`;
